numpy/lib/function_base.py

```diff
@@ -3447,6 +3465,60 @@ def percentile(a, q, axis=None, out=None,
     the median if ``q=50``, the same as the minimum if ``q=0`` and the
     same as the maximum if ``q=100``.
 
+    In the simplest case, where all weights are integers,
+    the `weights` argument can be seen as repeating the data in the sample.
+    For example, if a = [1,2,3] and weights = [1,2,3], this will be identical
+    to the case where there are no weights and a = [1,2,2,3,3,3], with the
+    percentiles interpolated in the cumulative probability space of the
+    expanded array.
+
+    The algorithm is illustrated here:
+
+    a = np.array([3, 5, 4])
+    q = [25, 50, 75]
+    axis = 0
+    weights = [1, 2, 3]
+
+    Upon expansion, and converting q to quantile space ([0, 1]):
+
+    value       3   4   4   4   5   5
+    cum_prob    0  .2  .4  .6  .8   1
+    q=0.25            4
+    q=0.5                 4
+    q=0.75                    4.75
+
+    returns [4, 4, 4.75]
+
+    In this method, the computation of weighted percentile uses
+    normalized cumulative weights, with some renormalization and edge-setting
+    to maintain consistency with the case where all weights are integers.
+
+    The cumulative probability bands associated with each value,
+    based on the weights:
+
+    value               3           4           5
+    weight              1           3           2  (total = 6)
+    cum. weight         .17         .67         1
+    prob_band         0 - .17   .33 - .67   .83 - 1  (.17-.33 and .67-.83 are
+                                                      transitions)
+    w slice             .17         .17         .17  (diff in upper bound,
+                                                      divided by weight)
+    lower_bound         0           .33         .83
+    upper_bound         .17         .67         1
+
+    Now subtract the bound values by the left-most w-slice value
+    new_lower_bd        0           .17         .67
+    new_upper_bd        0           .5          .83  --> used to renormalize
+    renormalized_lower  0           .2          .8
+    renormalized_upper  0           .6          1
+    new line up         3       3       4       4       5       5
+    quantile bounds     0       0       .2      .6      .8      1
+    q=0.25                                 4
+    q=0.5                                     4
+    q=0.75                                          4.75
+
```

Comment:
This documentation is going to render very badly. You might want to mark it as a code block or a table

Reply:
I'll gonna try putting `::` before it, and indenting the whole block, to see if it gets better.